packages/gitbook-docs/src/markdoc/schema/pageLink.ts

```diff
@@ -0,0 +1,23 @@
+import { Schema, Tag } from '@markdoc/markdoc';
+import { LinkAttr } from '../attributes';
+
+export type PageLinkProps<A> = {
+  readonly url: string;
+  readonly children: A;
+};
+
+export const pageLink: Schema = {
+  children: ['Link'],
+  attributes: {
+    url: {
+      type: LinkAttr,
+      required: true,
+    },
+  },
+  transform: (node, config) => {
+    const children = Array.from(node.walk())
+      .find(({ type }) => type === 'link')
+      ?.transformChildren(config);
```

Comment:
Does it work also if we render all the children?
```suggestion
    const children = node.transformChildren(config);
```

Reply:
Doing that, we are going to receive a `Link` tag as children of a `Paragraph`. And the `Paragraph` tag is the child of the `PageLink` tag.
```
{
  "$$mdtype": "Tag",
  "name": "Paragraph",
  "attributes": {},
  "children": [
    {
      "$$mdtype": "Tag",
      "name": "Link",
      "attributes": {
        "href": "/path/to/a/b"
      },
      "children": [
        "b.md"
      ]
    }
  ]
}
```